mount/upgrader.go

```diff
@@ -43,11 +47,32 @@ func Upgrade(underlying Mount, rootdir, initial string) (*Upgrader, error) {
 		return ret, nil
 	}
 
-	if initial != "" {
-		if _, err := os.Stat(initial); err == nil {
-			ret.transient = initial
-			return ret, nil
-		}
+	if _, err := os.Stat(ret.transientFilePath()); err == nil {
+		return ret, nil
+	}
+	// clean up the existing transient if it's gone bad
+	_ = os.Remove(ret.transientFilePath())
+
+	if initial == "" {
+		return ret, nil
+	}
+	// we don't want to rename the file given by the client -> simply copy it and use the upgrader's file naming scheme here.
+	src, err := os.Open(initial)
+	if err != nil {
+		return nil, err
+	}
+	defer src.Close()
+
+	dst, err := os.Create(ret.transientFilePath())
+	if err != nil {
+		return nil, err
+	}
+	defer dst.Close()
```

Comment:
I don't think we want a _fixed_ file per shard. This worked in your resumption test because you deleted the directory between shutdown and restart. Otherwise there's a 0-size file left behind which the DAG store thinks is a valid transient copy (because we don't do size or checksum checks at the moment), and it fails to initialise the shard. I think we need a safer approach, such that we don't expose the transient path (and it's a random file, so that it cannot be guessed) UNTIL we have finished fetching the mount the first time. Otherwise, the DAG store will save the transient path even if it hasn't finished downloading, and it will be corrupted on restart.